src/software/new_geom/point.h

```diff
@@ -0,0 +1,451 @@
+    /**
+     * Returns the distance between this Point and the given point p
+     *
+     * @return the distance
+     */
+    double distanceFromPoint(const Point &p) const;
+
+    /**
+     * Returns a new Vector from this Point
+     *
+     * @return A new vector from this Point
+     */
+    Vector toVector() const;
+
+    /**
+     * Returns the unit vector in direction of this point
+     *
+     * @return A unit vector in the direction of this Point, or a
+     * zero-vector if this Point is zero
+     */
+    Vector norm() const;
+
+    /**
+     * Returns a scaled normalized vector in the same direction as this
+     * Point
+     *
+     * @param length the desired length of the resultant vector
+     *
+     * @return a vector in the same direction as this Point and with the given length,
+     * or a zero-vector if this Point is zero
+     */
+    Vector norm(double length) const;
+
+    /**
+     * Returns a new Point that is this Point rotated counterclockwise by an angle
+     *
+     * @param rot the angle to rotate the Point
+     *
+     * @return the new Point rotated by rot
+     */
+    Point rotate(const Angle &rot) const;
+
+    /**
+     * Returns the direction of this Point
+     *
+     * @return the direction of this Point, in the range [-π, π], with 0 being
+     * the positive x direction, π/2 being up (positive y), etc. like on a standard x-y
+     * plane
+     *
+     *              +
+     *              | Y
+     *              |
+     *              |
+     *      +---------------+
+     *       -X     |      X
+     *              |
+     *              | -Y
+     *              +
+     */
+    Angle orientation() const;
+
+    /**
+     * Checks whether this Point contains NaN in either coordinate
+     *
+     * @return true if either coordinate is NaN, and false otherwise
+     */
+    constexpr bool isnan() const;
+
+    /**
+     * Checks whether this Point is close to another Point, where “close”
+     * is defined as 1.0e-9
+     *
+     * @param other the other point to check against
+     *
+     * @return true if the other point is within a distance of 1.0e-9, not inclusive
+     */
+    bool isClose(const Point &other) const;
+
+    /**
+     * Checks whether this Point is close to another Point
+     *
+     * @param other the other point to check against
+     *
+     * @param dist the distance to check against
+     *
+     * @return true if the other point is within the given distance (not inclusive) of
+     * this Point
+     */
+    bool isClose(const Point &other, double dist) const;
+
+    /**
+     * Assigns one Point to another
+     *
+     * @param other the Point whose value should be copied into this Point
+     *
+     * @return this Point
+     */
+    Point &operator=(const Point &other);
+
+   private:
+    /**
+     * The X coordinate of the Point. The variable name starts with an underscore to
+     * prevent name conflicts with its accessor function.
+     */
+    double _x;
```

Comment:
super duper nit: Generally if we have to add an underscore to a member variable so that it doesn't match member functions, we append it (ie. `x_` rather than `_x`).

(same for `_y`)